src/oic/oauth2/message.py

```diff
@@ -711,30 +722,30 @@ def verify(self, **kwargs):
         except KeyError:
             _allowed = {}
 
-        for (attribute, (typ, required, _, _, na)) in _spec.items():
+        for (attribute, cparam) in _spec.items():
             if attribute == "*":
                 continue
 
             try:
                 val = self._dict[attribute]
             except KeyError:
-                if required:
+                if cparam.required:
                     raise MissingRequiredAttribute("%s" % attribute)
                 continue
             else:
-                if typ == bool:
+                if cparam.type == bool:
```

Comment:
can be simplified to:

    if not cparam.type == bool and not val:
        if cparam.required:
            raise...
        continue